### PR TITLE
Added logic to show all errors from subschema validation for anyOf and oneOf

### PR DESCRIPTION
**Pull request by shelbys**
@jzogg and @y-tang: Code review, please for PLATAPI-3322

### CODE REVIEW

File: lib/attribute.js [hunk 1]
```diff
@@ -140,6 +153,22 @@ validators.anyOf = function validateAnyOf (instance, schema, options, ctx) {
...
+    errors.push.apply(errors, results[index].errors);
+  }
+  return {
+    errors: errors,
+    instance: instance,
+    propertyPath: ctx.propertyPath,
+    schema: schema
+  }
+
   if (!schema.anyOf.some(testSchema.bind(this, instance, options, ctx))) {
```
Comment (y-tang) [inline lib/attribute.js]: We should remove 172 to 177

**Comment by y-tang** [comment]
We might want to add some tests

**Comment by shelbys** [comment]
@y-tang - Re-review, please

File: lib/attribute.js (hunk 1 above)
Comment (shelbys) [inline lib/attribute.js]: Nice catch!

File: test/attributes.js [hunk 2]
```diff
@@ -532,8 +532,103 @@ describe('Attributes', function () {
...
+              prop1:{
+                enum:['3','4']
+              }
+            }
+          }
+        ]
+      };
+
+      describe('when instance is invalid', function(){
+        it('should return generic error message', function(){
```
Comment (y-tang) [inline test/attributes.js]: should _not_ return generic error message?

File: test/attributes.js [hunk 3]
```diff
@@ -532,8 +532,103 @@ describe('Attributes', function () {
...
+            prop1:{
+              enum:['2','3']
+            }
+          }
+        }
+      ]
+    };
+
+    describe('when instance is invalid', function(){
+      it('should return generic error message', function(){
```
Comment (y-tang) [inline test/attributes.js]: should _not_ return generic error message?

**Comment by y-tang** [comment]
:+1:

**Pull request merged**